src/app/commands/cmd_remove_layer.cpp

```diff
@@ -116,9 +116,16 @@ RemoveLayerCommand::RemoveLayerCommand()
 
 bool RemoveLayerCommand::onEnabled(Context* context)
 {
-  return context->checkFlags(ContextFlags::ActiveDocumentIsWritable |
-                             ContextFlags::HasActiveSprite |
-                             ContextFlags::HasActiveLayer);
+  if (!context->checkFlags(ContextFlags::ActiveDocumentIsWritable |
+                 ContextFlags::HasActiveLayer))
+    return false;
+
+  const ContextReader reader(context);
+  const Sprite* sprite(reader.sprite());
+  const Layer* layer = reader.layer();
+
+  return sprite && !((sprite->root()->layersCount() == 1) && (layer->parent() == sprite->root()));
+  
```

Comment:
This expression is confusing and I think we can simplify it checking for the true cases (intead of the inverse):
```suggestion
  return sprite && layer &&
    // We can remove all layers from non-root groups
    ((layer->parent() != sprite->root()) ||
     // Check that we are not removing the last layer in the sprite
     (sprite->root()->layersCount() > 1));  
```
But I saw that you wrote it originally in this way in 56726d2e8b0806fab54812001903a561ee8925dc and then reverted with 5b02460299129c6c6101c4d8bca11fabe9b923c8. Logically they do the same (by [De Morgan's law](https://en.wikipedia.org/wiki/De_Morgan%27s_laws)). Can you explain why did you change this? Did this last commit really fix what it says it fix? (because it should work exactly the same as the previous one).

Reply:
I believe I changed it when testing my code and I saw that checking the true cases was still allowing you to delete a group if it was the root, but I checked just now and the code is identical and works the same, so I must have done something wrong in testing, sorry.
I'll add your suggestions and push the code with comments.